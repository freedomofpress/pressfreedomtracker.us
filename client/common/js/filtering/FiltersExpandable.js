import React from 'react'
import classNames from 'classnames'


function FiltersExpandable({ filtersExpanded, children }) {
	return (
		<div className={classNames(
			'filters__expandable',
			{ 'filters__expandable--expanded': filtersExpanded }
		)}>
			{children}
		</div>
	)
}


export default FiltersExpandable
