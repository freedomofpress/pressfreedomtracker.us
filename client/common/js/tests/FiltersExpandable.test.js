import React from 'react';
import { shallow, mount } from 'enzyme';
import FiltersExpandable from '~/filtering/FiltersExpandable';
import {
    FILTERS_EXPANDED
} from '~/tests/factories/CategoriesFilterFactory';

describe('FiltersExpandable', () => {
    let filtersExpanded;

    beforeEach(() => {
        filtersExpanded = FILTERS_EXPANDED;
    });

    afterEach(() => {
        filtersExpanded = null;
    });

    it('should render without error in "debug" mode', () => {
        const component = shallow(
            <FiltersExpandable
                filtersExpanded={filtersExpanded}
            />
        );

        expect(component.debug()).toMatchSnapshot();
	});

	it('should render FiltersExpandable properly with filtersExpanded values', () => {
		const component = mount(
            <FiltersExpandable
                filtersExpanded={filtersExpanded}
            />
        );
        let expandedFilter = component.find('.filters__expandable.filters__expandable--expanded');
        expect(expandedFilter.length).toEqual(1);

        const unexpandedComponent = mount(
            <FiltersExpandable
                filtersExpanded={!filtersExpanded}
            />
        );
        expandedFilter = unexpandedComponent.find('.filters__expandable.filters__expandable--expanded');
        expect(expandedFilter.length).toEqual(0);
    });
});