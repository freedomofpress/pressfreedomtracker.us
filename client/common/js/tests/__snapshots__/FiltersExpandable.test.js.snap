// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersExpandable should render without error in "debug" mode 1`] = `"<div className=\\"filters__expandable filters__expandable--expanded\\" />"`;
